packages/shared/scope-hoisting/src/concat.js

```diff
@@ -203,9 +203,11 @@ function getUsedExports(
 
   let entry = bundle.getMainEntry();
   if (entry) {
-    for (let {asset, symbol} of bundleGraph.getExportedSymbols(entry)) {
+    for (let {asset, symbol, exportSymbol} of bundleGraph.getExportedSymbols(
+      entry,
+    )) {
       if (symbol) {
-        markUsed(asset, symbol);
+        markUsed(asset, exportSymbol);
```

Comment:
Was this a separate issue?

Reply:
No, I initially tried to extend the concat exclusion logic but then gave up because it's strictly less powerful than symbol propagation.

The issue here was that the `usedExports` map contained things like `$affe23$export$foo` which is not the exported name but the symbol name.